docs/001_develop/03_client-capabilities/010_filters/foundation-filters.mdx

```diff
@@ -0,0 +1,437 @@
+
+```json
+{
+  "dependencies": {
+    "@genesislcap/foundation-filters": "latest"
+  },
+}
+```
+
+## Available filters
+
+Here is a list of available filters in Genesis's `foundation-filters` package:
+
+* **nodeEnv**<!-- TODO: [nodeEnv](./docs/api/foundation-filters.nodeenvfilter.md) -->: This filter is used to match the current environment(production, development etc.) with the environment specified(passed as function param). It is useful for supporting environment-specific conditions ensuring that certain features are only available for targetted environment.
+* **percentage**<!-- TODO: [percentage](./docs/api/foundation-filters.percentagefilter.md) -->: The percentage filter in Genesis foundation filters checks if a randomly generated number between 0 and 100 is within a specified percentage threshold. This allows for setting probabilistic conditions, enabling features or elements to be displayed to a proportion of users based on the given percentage. This filter is useful for scenarios like A/B testing or gradually rolling out new features.
+* **timeWindow**<!-- TODO: [timeWindow](./docs/api/foundation-filters.timewindowfilter.md) -->: This filter is used to check whether the current date and time come in between the start and end period specified in the parameters.The filter returns true if the current date and time falls between the range, thus making it useful for time-sensitive conditions.
+* **urlTargetting**<!-- TODO: [urlTargeting](./docs/api/foundation-filters.urltargetingfilter.md) -->: It is used to filter content based on URL parts like scheme, path, host, or pattern matching. It allows to target specific URLs based on particular criteria, which can be useful for enabling or restricting features.
+* **userAgent**<!-- TODO: [userAgent](./docs/api/foundation-filters.useragentfilter.md) -->: It filters content based on user's browser, OS, devices or specified user agent string. It allows customized expereince according to specific environments, such as targetting mobile devices audience or restricting access on a particular browser.
+* **userTargeting**<!-- TODO: [userTargeting](./docs/api/foundation-filters.usertargetingfilter.md) -->: This filters content on basis of user related attributes like usernames, permissions and profiles. It aids in showing personalized content or restricting access to specific content depending on the soecified user conditions.
+
+
+## Usage
+
+Each filter is provided in two flavours:
+1. Functions
+2. Dependency Injection(DI)
+
+These flavours are explained in detail below with help of code examples
+<FiltersDemo />
+### As Function
+
+    ```typescript
+    import { timeWindowFilter } from '@genesislcap/foundation-filters';
+
+    @customElement({
+      name: 'my-element',
+      styles: css`
+        .container {
+          display: flex;
+          flex-direction: column;
+          gap: calc(var(--design-unit) * 3px);
+          padding: calc(var(--design-unit) * 4px);
+        }
+      `,
```

Comment:
Let's get rid of the styles here as they're not strictly necessary

Reply:
> Let's get rid of the styles here as they're not strictly necessary

ok I added styles as they were part of live example too so that when user copy this code he gets similar look

Reply:
removed as discussed